src/java.base/share/classes/java/security/Security.java

```diff
@@ -653,14 +653,7 @@ public static Provider getProvider(String name) {
         if (candidates == null || candidates.isEmpty())
             return null;
 
-        Object[] candidatesArray = candidates.toArray();
-        Provider[] result = new Provider[candidatesArray.length];
-
-        for (int i = 0; i < result.length; i++) {
-            result[i] = (Provider)candidatesArray[i];
-        }
-
-        return result;
+        return candidates.toArray(new Provider[0]);
```

Comment:
Is this called often enough to warrant creating a constant of `new Provider[0]` (benefits of this covered in the _Wisdom of the Ancients_ blog linked)?

Reply:
May be yes, may be not. I like `0`-sized array approach more.
1. It looks clearer and easier to read
2. It should be faster than original code anyway